packages/medusa-payment-stripe/src/helpers/stripe-base.js

```diff
@@ -99,49 +95,117 @@ class StripeBase extends AbstractPaymentService {
-   * @return {Promise<object>} Stripe customer
-   */
-  async createCustomer(customer) {
-    return await this.stripeProviderService_
-      .withTransaction(this.manager_)
-      .createCustomer(customer)
+    if (!customerId) {
+      return
+    }
+    return await this.stripe_.customers.retrieve(customerId)
   }
 
   /**
    * Creates a Stripe payment intent.
    * If customer is not registered in Stripe, we do so.
-   * @param {Cart} cart - cart to create a payment for
-   * @return {Promise<PaymentSessionData>} Stripe payment intent
+   * @param {Cart & PaymentContext} context - context to use to create a payment for
+   * @return {Promise<PaymentSessionResponse>} Stripe payment intent
    */
-  async createPayment(cart) {
+  async createPayment(context) {
     const intentRequestData = this.getPaymentIntentOptions()
+    const { id: cart_id, email, context: cart_context, collected_data, currency_code, amount, resource_id } = context
+
+    if (!isDefined(currency_code) || !isDefined(amount)) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_ARGUMENT,
+        'Unable to create a new payment, one of the "currency_code" or "amount" values is not provided. Both values are required.'
+      )
+    }
+
+    const intentRequest = {
+      description:
+        cart_context.payment_description ??
+        this.options_?.payment_description,
+      amount: Math.round(amount),
+      currency: currency_code,
+      metadata: { cart_id, resource_id },
+      capture_method: this.options_.capture ? "automatic" : "manual",
+      ...intentRequestData,
+    }
 
-    return await this.stripeProviderService_
-      .withTransaction(this.manager_)
-      .createPayment(cart, intentRequestData)
+    if (this.options_?.automatic_payment_methods) {
+      intentRequest.automatic_payment_methods = { enabled: true }
+    }
+
+    if (collected_data?.stripe_id) {
+      intentRequest.customer = collected_data.stripe_id
+    } else {
+      const stripeCustomer = await this.stripe_.customers.create({
+        email,
+      })
+
+      intentRequest.customer = stripeCustomer.id
+    }
+
+    const session_data = await this.stripe_.paymentIntents.create(
+      intentRequest
+    )
+
+    return {
+      session_data,
+			collected_data: {
+				customer: {
+					stripe_id: intentRequest.customer
+				}
+			}
+    }
   }
 
   async createPaymentNew(paymentInput) {
     const intentRequestData = this.getPaymentIntentOptions()
+    const { customer, currency_code, amount, resource_id, cart } = paymentInput
+    const { email } = customer ?? {}
+
+    const intentRequest = {
+      description:
+        cart?.context?.payment_description ??
+        this.options_?.payment_description,
+      amount: Math.round(amount),
+      currency: currency_code,
+      metadata: { resource_id },
+      capture_method: this.options_.capture ? "automatic" : "manual",
+      ...intentRequestData,
+    }
+
+    if (collected_data?.stripe_id) {
```

Comment:
thought: Where do you get `collected_data` from? Don't think you've defined it anywhere

Reply:
It is part of the type of object for the parameter

Reply:
But you are not destructuring it on line 162 as I would expect, so it is `undefined` right?

Reply:
you are right, I ve mixed up the methods in my head, this one is the new and will be merged In the existing one instead. I ll fix it

Reply:
included [here](https://github.com/medusajs/medusa/pull/2743/commits/5b30e2016c3ff51e2dc482d78f2f9f975a115326)